rclcpp_action/test/test_server.cpp

```diff
@@ -1316,6 +1312,9 @@ TEST_F(TestDeadlockServer, deadlock_while_succeed_and_canceled)
   send_goal_request(node_, uuid1_);
   std::thread t(&TestDeadlockServer::GoalSucceeded, this);
   rclcpp::sleep_for(std::chrono::milliseconds(50));
```

Comment:
why are we calling `sleep_for()`? isn't this extremly flaky?

Reply:
The aim is to ensure that this thread `std::thread t(&TestDeadlockServer::GoalSucceeded, this);` could be running first.

Reply:
My comment is because `sleep_for` cannot ensure that.

Reply:
I do agree with you that `sleep_for` cannot ensure that, but I think it works in most of the time. I just modified base on original test case.

Do you think we should improve `TestDeadlockServer` test cases in this PR?

Reply:
> I just modified base on original test case

That's true, sounds fair to keep it for the moment.